src/app/state/slices/api/gameboards.ts

```diff
@@ -48,8 +48,8 @@ export const assignGameboard = createAsyncThunk(
 
     // TODO think about whether this can be done in the back-end too?
     if (dueDate !== undefined) {
-      dueDate?.setHours(0, 0, 0, 0);
-      if (dueDate.valueOf() - today.valueOf() < 0) {
+      const modifiedDueDate = new Date(dueDate).setHours(0, 0, 0, 0);
```

Comment:
There are a couple if issues here:
1. `modifiedDueDate` is not very informative: what's the intention of this modification? Why do we need to set the hours, minutes, secs and millisecs to zero? That may give you a hint to rename this constant to something better.
2. The behaviour has now changed: the way this worked before your changes meant that `dueDate` would be changed after executing the `dueDate?.setHours(0, 0, 0, 0);` line, but now `dueDate` is never modified, because you've saved the changes in a different variable. But bear in mind that `dueDate` is used a few more times inside this function. With your changes, due date will now preserve its originals values for the hour, minutes, seconds and milliseconds. You need to review if this is the right behaviour, because it's now changed.

Reply:
I've renamed that to `startOfDayDueDate` and a new variable `dueDateInPast` to make it clearer what it's checking.

However on the second point, while the change doesn't exactly mimic the previous behaviour, I don't believe this changes any functionality as the actual calendar day remains the same, and due date doesn't care about the time. It was adjusted here just to consider that a same-day due date is treated as being in the past (so a student always has at least 1 day to complete assignments).

Reply:
Excellent, thank you very much!!